src/middlewared/middlewared/plugins/idmap.py

```diff
@@ -693,10 +714,33 @@ async def sid_to_name(self, sid_str):
         return rv
 
     @private
-    async def name_to_sid(self, name):
-        wb = await run([SMBCmd.WBINFO.value, '--name-to-sid', name], check=False)
+    async def unixid_to_sid(self, data):
+        """
+        Samba generates SIDs for local accounts that lack explicit mapping in
+        passdb.tdb or group_mapping.tdb with a prefix of S-1-22-1 (users) and
+        S-1-22-2 (groups). This is not returned by wbinfo, but for consistency
+        with what appears when viewed over SMB protocol we'll do the same here.
+        """
+        unixid = data.get("id")
+        id = IDType[data.get("id_type", "GROUP")]
+        wb = None
+        if id == IDType.USER:
+            wb = await run([SMBCmd.WBINFO.value, '--uid-to-sid', str(unixid)], check=False)
+        else:
+            wb = await run([SMBCmd.WBINFO.value, '--gid-to-sid', str(unixid)], check=False)
+
         if wb.returncode != 0:
-            self.logger.debug("wbinfo failed with error: %s",
-                              wb.stderr.decode().strip())
+            self.logger.warning("Could not convert [%d] to SID: %s",
+                                unixid, wb.stderr.decode().strip())
+            if WBCErr.DOMAIN_NOT_FOUND.err() in wb.stderr.decode():
+                is_local = await self.middleware.call(
+                    f'{"user" if id == IDType.USER else "group"}.query',
+                    [(f'{"uid" if id == IDType.USER else "gid"}', '=', unixid)],
```

Comment:
f-string seems feels odd

just `"uid" if id == IDType.USER else "gid"` would do it

Reply:
I agree.